chapters/annotations.tex

```diff
@@ -2052,24 +2055,56 @@ \subsection{Protection of Classes}\label{protection-of-classes}
   Same as \lstinline!Access.packageText! and additionally the class, or part of the class, can be copied.
 \end{enumerate}
 
-The \lstinline!access! annotation holds for the respective class and all classes
-that are hierarchically on a lower level, unless overridden by a
-\lstinline!Protection! annotation with \lstinline!access!.
-Overriding \lstinline!access=Access.hide! or \lstinline!access=Access.packageDuplicate!
-has no effect.
+The \lstinline!access! annotation holds for the respective class and all classes that are hierarchically on a lower level, unless overridden by a \lstinline!Protection! annotation with \lstinline!access!.
+Overriding \lstinline!access=Access.hide! or \lstinline!access=Access.packageDuplicate! has no effect.
 
 \begin{example}
 If the annotation is given on the top level of a package and at no other class in this package, then the \lstinline!access! annotation holds for all classes in this package.
 \end{example}
 
+Classes should not use other classes in ways that contradict this protection.
+Tools must ensure that protected contents are not shown, even if classes do not meet this requirement.
+
+\begin{example}
+For instance a class with \lstinline!Access.hide! should not be used in the diagram layer of a class with \lstinline!Access.diagram!, and there should not be hyperlinks to classes with \lstinline!Access.icon! (from classes with visible documentation).
+
+Consider the following invalid use of a class with \lstinline!Access.hide!:
+\begin{lstlisting}[language=modelica]
+package P
+  block MySecret
+    RealOutput y = time;
+    annotation(Protection(access = Access.hide));
+  end MySecret;
+  model M
+    MySecret mySecret;
+    Integrator integrator;
+  equation
+    connect(mySecret.y, integrator.u);
+    annotation(Protection(access = Access.diagram));
+  end M;
+end P;
```

Comment:
The problematic case of even less protection within the package should also be discussed:
```suggestion
  model M2
    MySecret mySecret;
    annotation(Protection(access = Access.packageDuplicate));
  end M2;
end P;
```

Reply:
There are too many open suggestions right now for also suggesting the text to go with it:
> Despite the protection level of \lstinline!P.M2!, a too must be cautious to not reveal the text with enough detail to tell that there is a reference to the class \lstinline!P.MySecret! (this is different compared to when such misuse of a protected class occurs outside of the package containing the protected class).

Reply:
I understand the concern, but one could also just have a text saying: "There is a hidden class MySecret" in the documentation, or having `//MySecret mySecret;` in the code.

(Technically using `MySecret` in an exposed class inside the protected package doesn't necessarily imply that there's a class called `MySecret`, as the package might have `import MySecret=Modelica.Blocks.Math.Gain;`)